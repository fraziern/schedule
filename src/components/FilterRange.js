import React, { Component, PropTypes } from "react";

class FilterRange extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="form-inline">
        <label htmlFor="startDate">Show Dates From</label>
        <input id="startDate" type="text" onChange={this.props.handleStartChange} value={this.props.startValue} onKeyPress={this.props.handleStartSubmit}/>
        <label htmlFor="endDate">To</label>
        <input id="endDate" type="text" placeholder="end of time" onChange={this.props.handleStopChange} value={this.props.stopValue} onKeyPress={this.props.handleStopSubmit}/>
      </div>
    );
  }
}

FilterRange.propTypes = {
  handleStartChange: PropTypes.func.isRequired,
  handleStopChange: PropTypes.func.isRequired,
  handleStartSubmit: PropTypes.func.isRequired,
  handleStopSubmit: PropTypes.func.isRequired,
  startValue: PropTypes.string.isRequired,
  stopValue: PropTypes.string
};

export default FilterRange;
