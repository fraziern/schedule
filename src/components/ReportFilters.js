import React, { PropTypes } from "react";
import FilterButton from "./FilterButton.js";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { updateReportFilter } from "../actions";

export function ReportFilters(props) {
  return (
    <div className="report-filterbar menubar">
      <div className="filter-buttons">
        <FilterButton filter="Year" label="1 Year" onClick={props.updateReportFilter} filterStatus={props.reportFilter}/>
        <FilterButton filter="9 Months" label="9 Months" onClick={props.updateReportFilter} filterStatus={props.reportFilter}/>
        <FilterButton filter="6 Months" label="6 Months" onClick={props.updateReportFilter} filterStatus={props.reportFilter}/>
        <FilterButton filter="3 Months" label="3 Months" onClick={props.updateReportFilter} filterStatus={props.reportFilter}/>
      </div>
    </div>
  );
}

ReportFilters.propTypes = {
  reportFilter: PropTypes.string.isRequired,
  updateReportFilter: PropTypes.func.isRequired
};

function mapStateToProps(state) {
  return { reportFilter: state.reports.reportFilter };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ updateReportFilter }, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(ReportFilters);
