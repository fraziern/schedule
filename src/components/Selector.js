import React, { Component } from "react";

class Selector extends Component {
  render() {
    const className = (this.props.selectorValue === "") ? "selector selector-empty" : "selector";

    return (
      <input className={className} type="text" name={this.props.assignment} disabled={this.props.cardDisabled} value={this.props.selectorValue} onChange={this.props.handleSelectorChange} onFocus={this.props.handleSelectorFocus}
      onBlur={this.props.handleSelectorBlur} onKeyPress={this.props.handleSelectorEnter}
      />);
  }
}

export default Selector;
