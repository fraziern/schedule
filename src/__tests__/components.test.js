import React from "react";
import FilterRange from "../components/FilterRange.js";
import Selector from "../components/Selector.js";
import { ReportFilters } from "../components/ReportFilters.js";
import renderer from "react-test-renderer";

describe("Components", function() {
  describe("<ReportFilters />", function() {
    it("renders correctly", function() {
      var myMock = jest.fn();
      let tree = renderer.create(
        <ReportFilters
          reportFilter="1 Year"
          updateReportFilter={myMock}
          />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("<Selector />", function() {
    it("renders correctly", function() {
      var myMock = jest.fn();
      let tree = renderer.create(
        <Selector
          selectorValue=""
          assignmentName=""
          isDisabled={false}
          handleSelectorChange={myMock}
          handleSelectorFocus={myMock}
          handleSelectorBlur={myMock}
          handleSelectorEnter={myMock}
          registerInput={myMock}
          />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("<FilterRange />", function() {
    it("renders correctly", function() {
      var myMock = jest.fn();
      var tree = renderer.create(
        <FilterRange
          handleStartChange={myMock}
          handleStopChange={myMock}
          handleStartSubmit={myMock}
          handleStopSubmit={myMock}
          startValue=""
           />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
