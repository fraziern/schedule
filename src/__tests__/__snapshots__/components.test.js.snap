exports[`Components <FilterRange /> renders correctly 1`] = `
<div
  className="form-inline">
  <label
    htmlFor="startDate">
    Show Dates From
  </label>
  <input
    id="startDate"
    onChange={[Function]}
    onKeyPress={[Function]}
    type="text"
    value="" />
  <label
    htmlFor="endDate">
    To
  </label>
  <input
    id="endDate"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="end of time"
    type="text"
    value={undefined} />
</div>
`;

exports[`Components <ReportFilters /> renders correctly 1`] = `
<div
  className="report-filterbar menubar">
  <div
    className="filter-buttons">
    <button
      className="btn btn-default btn-sm"
      onClick={[Function]}
      type="button">
      1 Year
    </button>
    <button
      className="btn btn-default btn-sm"
      onClick={[Function]}
      type="button">
      9 Months
    </button>
    <button
      className="btn btn-default btn-sm"
      onClick={[Function]}
      type="button">
      6 Months
    </button>
    <button
      className="btn btn-default btn-sm"
      onClick={[Function]}
      type="button">
      3 Months
    </button>
  </div>
</div>
`;

exports[`Components <Selector /> renders correctly 1`] = `
<input
  className="form-control selector selector-empty"
  disabled={false}
  name=""
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyPress={[Function]}
  type="text"
  value="" />
`;
